### PR TITLE
Jpnlang 0006 user service swagger

### PR DESCRIPTION
**Pull request by shoaib812**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,2 @@
-# jpn-lang-api
\ No newline at end of file
+# jpn-lang-api
+# http://localhost:8082/swagger-ui/index.html
```
Comment (Mohdzaved2434) [inline README.md]: you should add the description, what is the use of this url?

File: README.md [hunk 2]
```diff
@@ -1 +1,2 @@
-# jpn-lang-api
\ No newline at end of file
+# jpn-lang-api
```
Comment (Mohdzaved2434) [inline README.md]: Can you update correct project name?

File: src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java [hunk 3]
```diff
@@ -3,25 +3,55 @@
...
+                            description = "Success",
+                            responseCode = "200"
+                    ),
+                    @ApiResponse(
+                            description = "Data Not Found",
+                            responseCode = "404"
+                    )
+            }
+    )
+    @PostMapping(value = "/employees", consumes = "application/json", produces = "application/json")
```
Comment (Mohdzaved2434) [inline src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java]: why did you add path in post mapping annotation here?
because now it has become 
http://localhost:8082/employees/employees

for post api

Comment (shoaib812) [reply src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java]: i have deleted this line.

File: README.md (hunk 1 above)
Comment (shoaib812) [reply README.md]: updated project name.

File: src/main/java/com/jpn/lang/config/SwaggerConfig.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.annotations.info.Contact;
+import io.swagger.v3.oas.annotations.info.Info;
+import io.swagger.v3.oas.annotations.info.License;
+import io.swagger.v3.oas.annotations.servers.Server;
+
+@OpenAPIDefinition(
+    info = @Info(
+            title = "JPN LANG",
```
Comment (Mohdzaved2434) [inline src/main/java/com/jpn/lang/config/SwaggerConfig.java]: I think project name is nihongo-api?

File: src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java [hunk 5]
```diff
@@ -3,25 +3,85 @@
...
+    @Operation(
+            tags = "ADD Employees",
+            description = "Add Employees",
+            responses = {
+                    @ApiResponse(
+                            description = "Success",
+                            responseCode = "200"
+                    ),
+                    @ApiResponse(
+                            description = "Data Not Found",
```
Comment (Mohdzaved2434) [inline src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java]: Will you get this error in case of post request?

File: README.md [hunk 6]
```diff
@@ -1 +1,10 @@
-# jpn-lang-api
\ No newline at end of file
+# nihongo-api
+# http://localhost:8082/swagger-ui/index.html
+
```
Comment (Mohdzaved2434) [inline README.md]: you should update like this.

Swagger URL
http://localhost:8082/swagger-ui/index.html

File: README.md [hunk 7]
```diff
@@ -1 +1,10 @@
...
+# What is swagger-ui 
+- Swagger-ui is a tool that provides a visual representation of APIs and their documentation.
+- It allows users to visualize and interact with API resources without needing to know the implementation logic.
```
Comment (Mohdzaved2434) [inline README.md]: you should remove all these things?

**Comment by Mohdzaved2434** [comment]
Can you check message of your last commit, is this correct format?
push swagger 

did you run sonarqube before pushing the code?

File: README.md (hunk 2 above)
Comment (shoaib812) [reply README.md]: i have already updated the project name.

File: src/main/java/com/jpn/lang/config/SwaggerConfig.java (hunk 4 above)
Comment (shoaib812) [reply src/main/java/com/jpn/lang/config/SwaggerConfig.java]: I have changed the project name.

File: src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java (hunk 5 above)
Comment (shoaib812) [reply src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java]: i do not get this error on browser.
i get only empty array.

File: README.md (hunk 6 above)
Comment (shoaib812) [reply README.md]: updated

File: README.md (hunk 7 above)
Comment (shoaib812) [reply README.md]: removed

File: src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java
Comment (Mohdzaved2434) [inline src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java:1]: You did not delete this package? I can still see jpn/lang in package

File: src/main/java/com/nihongo/NihonGoApi.java [hunk 8]
```diff
@@ -1,14 +1,14 @@
-package com.jpn.lang;
```
Comment (Mohdzaved2434) [inline src/main/java/com/nihongo/NihonGoApi.java:1]: Can you also make com.nihongo from nihongo in package folder name?

File: src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java (hunk 5 above)
Comment (Mohdzaved2434) [reply src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java]: Thats what I am saying you will not get this error in case of post API. Because you are not modifying or deleting existing object. instead you are creating new object. So you can remove Data not found response from post API.

**Comment by Mohdzaved2434** [comment]
> Can you check message of your last commit, is this correct format? Next time you should use correct format push swagger
> 
> did you run sonarqube before pushing the code?

@shoaib812 did you run sonarqube?

File: src/main/java/com/nihongo/NihonGoApi.java [hunk 9]
```diff
@@ -1,14 +1,14 @@
...
+package com.nihongo;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
-public class JpnLangApiDemo {
+public class NihonGoApi {
```
Comment (Mohdzaved2434) [inline src/main/java/com/nihongo/NihonGoApi.java]: Can you rename this class like this NihongoAPI? this looks better

File: src/main/java/com/jpn/lang/config/SwaggerConfig.java (hunk 4 above)
Comment (shoaib812) [reply src/main/java/com/jpn/lang/config/SwaggerConfig.java]: already changed

**Review by Mohdzaved2434 — APPROVED** [review]

File: src/main/java/com/nihongo/NihonGoApi.java (hunk 9 above)
Comment (shoaib812) [reply src/main/java/com/nihongo/NihonGoApi.java]: i have changed

File: src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java
Comment (shoaib812) [reply src/main/java/com/jpn/lang/controller/EmployeeControllerApi.java:1]: i have deleted.
